lib/td/client/api/job.rb

```diff
@@ -287,6 +248,114 @@ def query(q, type=:hive, db=nil, result_url=nil, priority=nil, retry_limit=nil,
 
   private
 
+  def validate_content_length(response, current_total_chunk_size)
+    if expected_size = response.header['Content-Range'][0]
+      expected_size = expected_size[/\d+$/].to_i
+    elsif expected_size = response.header['Content-Length'][0]
+      expected_size = expected_size.to_i
+    end
+
+    if expected_size.nil?
+    elsif current_total_chunk_size < expected_size
+      # too small
+      raise EOFError, 'httpclient IncompleteError'
```

Comment:
Previous code raised proprietary exception named IncompleteError but its original error is EOFError raised by ext/openssl but httpclient.gem ignores it. Because such proprietary exception takes additional cost to understand, I simply raised original exception.
Anyway I added comment.